src/decorator.rb

```diff
@@ -0,0 +1,34 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :name
+
+  def initialize(name)
+    super()
+    @name = name
+  end
+
+  def correct_name
+    @name.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @name.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name.slice(0, 10)
+    else
+      @Nameable.correct_name
+    end
```

Comment:
- [ ] Hello, Please kindly notice that you have a typo here. this is the incorrect variable name. you have defined the `@name`  instance variable which is what you should be using here. please make this change.